docs/plugins.md

```diff
@@ -0,0 +1,72 @@
+# Plugins
+
+![Plugins](assets/plugins.png)
+
+Plugins allow you to extend Argo Workflows to add new capabilities.
+
+* You don't need to learn Golang, you can write in any language, including Python.
+* Simple: a plugin just responds to RPC HTTP requests.
+* You can iterate quickly by changing the plugin at runtime.
+* You can get your plugin running today, no need to wait 3-5 months for an Argo software release.
+
+[Executor plugins](executor_plugins.md) written and installed by both users and admins.
+
+## Configuration
+
+Plugins are disabled by default. To enable them, start the controller with `ARGO_PLUGINS=true`, e.g.
```

Comment:
Just thinking it out: an allowlist would solve for the situation where an untrusted user may _add_ a plugin ConfigMap, but cannot _modify_ an existing plugin ConfigMap. 

If the malicious user could modify an existing plugin ConfigMap, they'd just add malicious code to the existing plugin rather than creating a new malicious plugin.

Reply:
I'm not sure an allow/deny list is needed. Users can add and remove plugins by creating and deleting config maps.

What _might_ be useful is some way to only enable plugins for some workflows, e.g. by label selector.